runtime/jcl/common/jclcinit.c

```diff
@@ -773,8 +771,39 @@ initializeRequiredClasses(J9VMThread *vmThread, char* dllName)
 					return 1;
 				}
 				j9str_printf(PORTLIB, propNameBuffer, sizeof(ADDMODS_PROPERTY_BASE) + indexLen, ADDMODS_PROPERTY_BASE "%zu", vm->addModulesCount);
-				Trc_JCL_initializeRequiredClasses_addAgentModuleSetProperty(vmThread, propNameBuffer);
-				vmFuncs->addSystemProperty(vm, propNameBuffer, AGENT_MODULE_NAME, J9SYSPROP_FLAG_NAME_ALLOCATED);
+				Trc_JCL_initializeRequiredClasses_addAgentModuleSetProperty(vmThread, propNameBuffer, moduleName);
+				vmFuncs->addSystemProperty(vm, propNameBuffer, moduleName, J9SYSPROP_FLAG_NAME_ALLOCATED);
+			}
+		}
+	}
+
+	if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_LOAD_HEALTHCENTER_MODULE)
+		&& (NULL != vm->modulesPathEntry->extraInfo))
+	{
+		const char *moduleName = vm->jimageIntf->jimagePackageToModule(
+				vm->jimageIntf, (UDATA) vm->modulesPathEntry->extraInfo,
+				"com/ibm/java/diagnostics/healthcenter/agent");
+		if (NULL != moduleName) {
+			J9VMSystemProperty *systemProperty = NULL;
+			Trc_JCL_initializeRequiredClasses_addAgentModuleEntry(vmThread, moduleName);
+			/* Handle the case where there are no user-specified modules. In this case, there
+			 * is typically one system property but no user-set "add-modules" arguments.
+			 */
+			if ((0 == vm->addModulesCount)
```

Comment:
Since adding the agent module above doesn't increment addModulesCount, both of the cases below won't work if -Xhealthcenter is used in combination.

Reply:
Fixed in 5cd45ce0fea3.